File: fe/src/app/models/task.ts
import {User} from 'src/app/models/user';
import {Project} from 'src/app/models/project';
export class Task{

    idTask:number;
    idCreatedBy:number;
    code:string;
    priority:string;
    status:string;
    description:string;
    createDate:Date;
    updateDate:Date;
    dueDate:Date;
    estimationDate:Date;
    taskUser:User;
    taskProject:Project;
    static cloneBase(task:Task):Task {
        const clonedTask:Task = new Task();
        clonedTask.idTask = task.idTask;
        clonedTask.idCreatedBy = task.idCreatedBy;
        clonedTask.code = task.code;
        clonedTask.priority = task.priority;
        clonedTask.status = task.status;
        clonedTask.description = task.description;
        clonedTask.createDate = task.createDate;
        clonedTask.updateDate = task.updateDate;
        clonedTask.dueDate = task.dueDate;
        clonedTask.estimationDate = task.estimationDate;
        clonedTask.taskUser = task.taskUser;
        clonedTask.taskProject = task.taskProject;
        return clonedTask;
    }
}

File: be/src/main/java/com/be/DTO/TaskMain/Task.java
package com.be.DTO.TaskMain;

import lombok.*;

import java.time.LocalDate;
import java.util.SortedSet;
import java.util.TreeSet;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
//@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "idTask",scope = Task.class)
public class Task {

    private int idTask;
    private int idCreatedBy;
    private int code;
    private String name;
    private String priority;
    private String status;
    private String description;
    private LocalDate createDate;
    private LocalDate updateDate;
    private LocalDate dueDate;
    private LocalDate estimationDate;

    private SortedSet<Comment> taskComments=new TreeSet<>();
    private User taskUser;
    private Project taskProject;

}
